embedded/sql/engine.go

```diff
@@ -1109,16 +1111,19 @@ func DecodeValue(b []byte, colType SQLValueType) (TypedValue, int, error) {
 		}
 	case IntegerType:
 		{
-			if vlen > 8 {
+			if vlen != 8 {
 				return nil, 0, ErrCorruptedData
 			}
 
-			buff := [8]byte{0}
+			buff := [8]byte{}
 			copy(buff[8-vlen:], b[voff:voff+vlen])
 			v := binary.BigEndian.Uint64(buff[:])
```

Comment:
Since 8 bytes are now required, the extra buffer is no longer needed:

```suggestion
			v := binary.BigEndian.Uint64(b[voff:])
```